platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/sync/config/SyncConfig.java

```diff
@@ -18,28 +18,47 @@
  *                                        the sync-as-a-protocol algorithm
- * @param syncProtocolPermitCount         the number of permits to use when running the sync-as-a-protocol algorithm
- * @param syncProtocolHeartbeatPeriod     the period at which the heartbeat protocol runs when the sync-as-a-protocol
- *                                        algorithm is active (milliseconds)
+ * @param syncProtocolPermitCount         the number of permits to use when running the sync algorithm
+ * @param onePermitPerPeer                if true, allocate exactly one sync permit per peer, ignoring
+ *                                        {@link #syncProtocolPermitCount()}. Otherwise, allocate permits according to
+ *                                        {@link #syncProtocolPermitCount()}.
+ * @param syncProtocolHeartbeatPeriod     the period at which the heartbeat protocol runs when the sync algorithm is
+ *                                        active (milliseconds)
  * @param hashOnGossipThreads             if true, hash events on gossip threads. If false, events are hashed on the
  *                                        event intake thread.
  * @param waitForEventsInIntake           if true, then a node will not sync again with the same peer until all
  *                                        previously received events from that peer have fully traversed the intake
  *                                        pipeline
  * @param criticalQuorumEnabled           if true, use the critical quorum heuristic when deciding if we should sync
  *                                        with a peer
+ * @param filterLikelyDuplicates          if true then do not send events that are likely to be duplicates when they are
+ *                                        received by the peer
+ * @param nonAncestorFilterThreshold      ignored if {@link #filterLikelyDuplicates} is false. For each event that is
+ *                                        not a self event and is not an ancestor of a self event, we must know about
+ *                                        the event for at least this amount of time before the event is eligible to be
+ *                                        sent
+ * @param syncKeepalivePeriod             send a keepalive message every this many milliseconds when reading events
+ *                                        during a sync
+ * @param maxSyncTime                     the maximum amount of time to spend syncing with a peer, syncs that take
+ *                                        longer than this will be aborted
  */
 @ConfigData("sync")
 public record SyncConfig(
         @ConfigProperty(defaultValue = "25") int syncSleepAfterFailedNegotiation,
         @ConfigProperty(defaultValue = "17") int syncProtocolPermitCount,
+        @ConfigProperty(defaultValue = "true") boolean onePermitPerPeer, // TODO turn this off prior to merge
```

Comment:
Leaving an open comment here as a reminder to remove the TODO

Reply:
reverted